src/entity.ts

```diff
@@ -372,7 +435,7 @@ export namespace entity {
       }
 
       case 'integerValue': {
-        return Number(value);
+        return decodeIntegerValue(valueProto, integerTypeCastOptions);
```

Comment:
This is definitely a more robust option, but was just returning the DS.Int type considered?

Reply:
Sorry, I mixed up Datastore's "Int" with Spanner's: https://github.com/googleapis/nodejs-spanner/blob/f8ad0ece28e99dd9c6d7e985de56bd1b376f4712/src/codec.ts#L139 -- for simplicity's sake, would it be worth considering using that format instead? When a value is out of bounds, it will throw. The value is accessible via the "value" property in string form.

Reply:
* I do like the option of ability to still access the value on default (without extra call) in case of value is out of balance.
* it will definitely be a breaking change
* it will add complexity to the user's code
   will need to `try=>catch` for `valueOf` and if `error` still pass to custom `typeCastFunction`.
* Is it overkill to combine these two approaches
     * return Spanner's style object for `integerValue`.
     * for `valueOf` on default throw in case of out of bounds.
     * able to pass integerTypeCastFunction and `valueOf` will return the custom function result.

Reply:
WDYT?

Reply:
Did we ever make progress on this, maybe in another chat?

Reply:
Quick summary
Would like to hear the thoughts on the approach
1. Throw on default, let user pass integerTypeCastFunction (currently implemented in this PR)
2. Return an object `Integer` where 
    * `#valueOf` would try to convert to `Number` and throw if out of bounds
    * `#value` returns `string` representation of the value
3. Combine tow approaches above 
    * do return an `Integer` object 
    * let user provide integerTypeCast function to be used by `#valueOf`
    * still return string via `#value`

Reply:
My vote would go for #&shy;3. Ping to @callmehiphop and @bcoe for more votes.

Reply:
My vote is starting simple with option 1, as a breaking change, and we can move towards an object representation in the future, if there's good reason.